mgnify_pipelines_toolkit/analysis/amplicon/amplicon_utils.py

```diff
@@ -0,0 +1,183 @@
+
+from collections import defaultdict
+import os
+import subprocess
+
+from mgnify_pipelines_toolkit.constants.regex_ambiguous_bases import _AMBIGUOUS_BASES_DICT, _AMBIGUOUS_BASES_DICT_REV
+
+def split_dir_into_sample_paths(_DIR):
+
+    file_list = os.listdir(_DIR)
+    file_list = [ file for file in file_list if '.fastq' in file and ('_1' in file or '_2' in file) ]
+    sample_set = set()
+    [ sample_set.add(f"{_DIR}/{file.split('_')[0]}") for file in file_list ]
+    sample_list = sorted(list(sample_set))
+
+    return sample_list
+
+def get_read_count(read_path, type='fastq'):
+
+    cmd = []
+    
+    if type == 'fastq':
+        cmd = [
+            'zgrep',
+            '-c',
+            '^@',
+            read_path
+        ]
+
+    elif type == 'fasta':
+        cmd = [
+            'grep',
+            '-c',
+            '^>',
+            read_path
+        ]
+
+    proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    stdout, stderr = proc.communicate()
+
+    read_count = int(stdout.strip())
+
+    return read_count
+
+def build_cons_seq(cons_list, read_count, cons_threshold=0.80, do_not_include=[], counter=1):
```

Comment:
This `do_not_include=[]` will introduce bugs in the code -> https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Reply:
Ahaaa I did not know this thank you, I'll make the default value Null like they recommend then